kernel/src/process.rs

```diff
@@ -319,7 +366,10 @@ pub enum FaultResponse {
     /// Attempt to cleanup and restart the app which caused the fault. This
     /// resets the app's memory to how it was when the app was started and
     /// schedules the app to run again from its init function.
-    Restart,
+    ///
+    /// The provided restart policy is used to determine whether to reset the
+    /// app, and can be specified on a per-app basis.
+    Restart(&'static dyn ProcessRestartPolicy),
```

Comment:
Just noting that this is going to add two words of memory to the kernel, regardless of whether this variant is used.

Reply:
It will add two words to the process, which we charge to the process's memory region.